Email improvements

alisonrclarke opened the pull request:
* Added extra button when adding review message for tables, to choose whether or not to send an email (Fixes #86)
 * Added extra button for uploaders and reviewers to enable them to send a message to all uploaders and reviewers, optionally including the status and messages for all tables (Fixes #86)
 * Added extra button for uploaders to allow them to notify the coordinator when all tables have passed (Fixes #93)
 * Send an email to coordinators when a submission is created (Fixes #99)

alisonrclarke committed:
Add "Save (no email)" button to reviewer screen

Currently this just determines whether to send that particular message.

alisonrclarke committed:
Replace 'notify reviewer' with 'notify participants' available to all participants

Replaced upload email with a submission_status email containing table status.

alisonrclarke committed:
Add test for getting participants with multiple roles

alisonrclarke committed:
Fixed test

 * Updated button id
 * Avoid race conditions when changing table status (only waiting if new status is not immediately visible)

alisonrclarke committed:
Added functionality to email coordinators when tables have all passed

alisonrclarke committed:
Updated buttons to use HEPData theme colours rather than bootstrap defaults

alisonrclarke committed:
Added checks for button toggles

alisonrclarke committed:
Fixed link error

alisonrclarke committed:
Send email to coordinators when a new submission has been created

alisonrclarke committed:
Merge branch 'master' into email-improvements

alisonrclarke committed:
Merge branch 'master' into email-improvements

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -84,50 +87,127 @@ def send_new_review_message_email(review, message, user):
                                        link=site_url + "/record/{0}"
                                        .format(recid))
 
+        if participant.role == 'uploader' and not reviewers_notified:
```
Should this be 'reviewer' instead of 'uploader'?

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -301,3 +381,31 @@ def notify_publication_update(hepsubmission, record):
                            message_body)
+
+
+def notify_submission_created(hepsubmission, record, coordinator_id, uploader, reviewer):
```
`hepsubmission` is not used in this function.

GraemeWatt commented on hepdata/modules/records/templates/hepdata_records/components/notify-coordinator-widget.html:
```diff
@@ -0,0 +1,73 @@
+                        </div>
+                    </div>
+                    <div id="coordinator-notification-sent" style="display:none">
+                        <h4>Emails sent.</h4>
```
Change to "Email sent." since only one email.

GraemeWatt commented on hepdata/modules/theme/templates/hepdata_theme/email/created.html:
```diff
@@ -0,0 +1,28 @@
+
+{% block email_content %}
+
+    Dear all,
```
This should be "Dear {{ name }},".

GraemeWatt commented on hepdata/modules/theme/templates/hepdata_theme/email/submission_status.html:
```diff
@@ -0,0 +1,86 @@
+    <p>Hi {{ name }},</p>
+
+    {% if invite_token %}
+        <p>You have been added as {{ role }} for a HEPData record. You can only review the article once you access the link below <strong>after logging in to HEPData</strong></p>
```
Change "review the article" to "view the article" in case the email is sent to another Uploader who has not already claimed permissions.

GraemeWatt commented on hepdata/modules/theme/templates/hepdata_theme/email/created.html:
```diff
@@ -0,0 +1,28 @@
+    Dear all,
+
+    <p>A new submission <a href="{{ link }}" style="color: #9b59b6">{{ title }}</a> has been created by {{ actor }}.</p>
+
```
It would be nice to include the collaboration name at the end of the sentence, e.g. "for {{ collaboration }}", where `collaboration` is taken from `CoordinatorRequest` with `approved=True` and `id` set to `coordinator_id`.  For an Admin user, there might not be an entry in `CoordinatorRequest`, so the sentence should be unchanged.

GraemeWatt commented on hepdata/modules/theme/templates/hepdata_theme/email/passed_review.html:
```diff
@@ -0,0 +1,23 @@
+        <p style="padding-left: 20px; font-weight: lighter; white-space: pre-wrap;">{{ message }}</p>
+    {% endif %}
+
+    <p>The submission is now ready to be finalised, which can be done from your <a href="{{ dashboard_link }}">HEPData dashboard</a>.</p>
```
Similarly, it would be nice to write "The submission is now ready to be finalised for {{ collaboration }}" if the collaboration name can be taken from `CoordinatorRequest`.

GraemeWatt commented on hepdata/modules/email/api.py:
```diff
@@ -84,50 +87,127 @@ def send_new_review_message_email(review, message, user):
+    if coordinator_profile:
+        name = coordinator_profile.full_name
+
+    message_body = render_template('hepdata_theme/email/passed_review.html',
```
Need to pass `version=version` to `passed_review.html` template.

GraemeWatt commented on hepdata/modules/records/views.py:
```diff
@@ -147,21 +147,48 @@ def submit_question(recid):
+            message=message
+        )
 
         submission.reviewers_notified = True
```
I think this line should be omitted.

GraemeWatt reviewed (changes requested):
This looks great!  I've tested and everything works well (after rebuilding the JS).  I spotted some minor issues and I've made comments in the relevant code.

alisonrclarke committed:
Merge branch 'master' into email-improvements

alisonrclarke committed:
Fixes following review

alisonrclarke committed:
Remove unnecessary update

Looks like this was a copy and paste error - we don't need to update
reviewers_notified when notifying the coordinator (so we won't need the
submission either)

GraemeWatt committed:
email: tweaks to inclusion of collaboration

* Check for approved request when getting collaboration.
* Add collaboration also to Uploader/Reviewer invitation email.
* Move mention of collaboration in passed_review.html.

GraemeWatt reviewed (approved):
Thanks for addressing my review comments.  I made some minor tweaks:

* Added `approved=True` when getting collaboration name.  There's one entry in the production database (`user=24`) which has two entries in `CoordinatorRequest`, one with `collaboration="CMS"` (`approved=False`) and one with `collaboration="CMS Heavy-Ion Physics"` (`approved=True`).  We need to make sure that only the latter is returned.
* In `passed_review.html` I prefer to move the collaboration to after "The submission is now ready to be finalised", because it is the Coordinator who needs to take action on behalf of the collaboration not the Reviewer.
* I included the collaboration also in the invitation email to Uploaders/Reviewers.
